rmw_cyclonedds_cpp/src/rmw_node.cpp

```diff
@@ -490,6 +508,33 @@ static void get_entity_gid(dds_entity_t h, rmw_gid_t & gid)
   convert_guid_to_gid(guid, gid);
 }
 
+static std::map<std::string, std::vector<uint8_t>> parse_user_data(const dds_qos_t * qos)
+{
+  void * ud;
+  size_t udsz;
+  if (!dds_qget_userdata(qos, &ud, &udsz)) {
+    std::map<std::string, std::vector<uint8_t>> emptymap;
```

Comment:
@eboasson nit: I'd think any compiler would be smart enough to copy-elide the return value, but

```c++
std::map<std::string, std::vector<uint8_t>> map;
void * ud;
size_t udsz;
if (dds_qget_userdata(qos, &ud, &udsz)) {
  std::vector<uint8_t> udvec(static_cast<uint8_t *>(ud), static_cast<uint8_t *>(ud) + udsz);
  dds_free(ud);
  map = rmw::impl::cpp::parse_key_value(udvec);
}
return map;
```
would make it simpler and slightly clearer.